homeassistant/components/nextcloud/entity.py

```diff
@@ -1,26 +1,29 @@
 """Base entity for the Nextcloud integration."""
-from homeassistant.helpers.entity import Entity
-from homeassistant.helpers.typing import StateType
 
-from .const import DOMAIN
+from typing import Any
 
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-class NextcloudEntity(Entity):
-    """Base Nextcloud entity."""
+from .coordinator import NextcloudDataUpdateCoordinator
 
-    _attr_icon = "mdi:cloud"
 
-    def __init__(self, item: str) -> None:
-        """Initialize the Nextcloud entity."""
-        self._attr_name = item
+class NextcloudEntity(CoordinatorEntity[NextcloudDataUpdateCoordinator]):
+    """Basis Nextcloud entity."""
+
+    icon = "mdi:cloud"
+
+    def __init__(self, coordinator: NextcloudDataUpdateCoordinator, item: str) -> None:
+        """Initialize the Nextcloud sensor."""
+        super().__init__(coordinator)
         self.item = item
-        self._state: StateType = None
+        self._attr_name = item
 
     @property
-    def unique_id(self):
-        """Return the unique ID for this sensor."""
-        return f"{self.hass.data[DOMAIN]['instance']}#{self.item}"
+    def data(self) -> dict[str, Any]:
+        """Return Nextcloud data."""
+        return self.coordinator.data
```

Comment:
I am not sure that this property is useful.
I think it is easy to adjuste each platform.